app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -773,10 +810,30 @@ public boolean onBackPressed() {
             return true; // no code here, as previous() was used in the if
         }
 
-        // That means that we are on the start of the stack,
+        // If we passed previous check at the top `player.getPlayQueue().previous()`
+        // then we can be sure that queue have no more streams in the history.
+        // If the user chose BACK_BEHAVIOR_PREV_PLAYLIST behavior
+        // we have to minimize the player into the mini player here
+        if (behavior == PlayerHelper.BackBehavior.BACK_BEHAVIOR_PREV_PLAYLIST) {
+            bottomSheetBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
+            return true;
+        }
+
+        // That means that we are on the start of the stack
         if (stack.size() <= 1) {
             restoreDefaultOrientation();
-            return false; // let MainActivity handle the onBack (e.g. to minimize the mini player)
+
+            if (behavior == PlayerHelper.BackBehavior.BACK_BEHAVIOR_HIDE_OR_CLOSE) {
```

Comment:
Why is "BACK_BEHAVIOR_HIDE_OR_CLOSE" here and not just after "BACK_BEHAVIOR_HIDE"? Shouldn't the two act the same way, with the only difference that "BACK_BEHAVIOR_HIDE_OR_CLOSE" should close the player if nothing is playing?

Reply:
The idea is to have a backstack working (like in BACK_BEHAVIOR_PREV_ANY) but to allow to close mini player completely. With BACK_BEHAVIOR_PREV_ANY the mini player will be at the bottom even when video is paused. With this option you're able to hide it in this case.

That's why it is the last check and it is inside `f (stack.size() <= 1) {`, which guarantees that backstack is empty and nothing can be played.